File: PrismaApi/PrismaApi.Domain/Dtos/NodeStyleDtos.cs
using System;
using System.Text.Json.Serialization;

namespace PrismaApi.Domain.Dtos;

public class NodeStyleDto
{
    [JsonPropertyName("id")]
    public Guid Id { get; set; } = Guid.NewGuid();
    [JsonPropertyName("node_id")]
    public Guid NodeId { get; set; }
    [JsonPropertyName("x_position")]
    public double XPosition { get; set; }
    [JsonPropertyName("y_position")]
    public double YPosition { get; set; }
}

public class NodeStyleIncomingDto : NodeStyleDto
{
}

public class NodeStyleOutgoingDto : NodeStyleDto
{
}

File: PrismaFastApi/src/dtos/node_style_dtos.py
import uuid
from pydantic import BaseModel, Field, field_validator

from src.constants import DtoConstants


class NodeStyleDto(BaseModel):
    id: uuid.UUID = Field(default_factory=uuid.uuid4)
    node_id: uuid.UUID
    x_position: float = 0.0
    y_position: float = 0.0

    @field_validator("x_position")
    @classmethod
    def x_position_validator(cls, value: float) -> float:
        return round(value, DtoConstants.DECIMAL_PLACES.value)

    @field_validator("y_position")
    @classmethod
    def y_position_validator(cls, value: float) -> float:
        return round(value, DtoConstants.DECIMAL_PLACES.value)


class NodeStyleIncomingDto(NodeStyleDto):
    pass


class NodeStyleOutgoingDto(NodeStyleDto):
    pass
